src/main/java/uk/gov/companieshouse/api/accounts/interceptor/ClosedTransactionInterceptor.java

```diff
@@ -1,56 +1,53 @@
 import uk.gov.companieshouse.api.accounts.transaction.Transaction;
-import uk.gov.companieshouse.api.accounts.transaction.TransactionManager;
 import uk.gov.companieshouse.api.accounts.transaction.TransactionStatus;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
 
 @Component
 public class ClosedTransactionInterceptor extends HandlerInterceptorAdapter {
 
-    @Autowired
-    private TransactionManager transactionManager;
+    private static final Logger LOGGER = LoggerFactory
+        .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
 
     /**
-     * Pre handle method to validate the request before it reaches the controllers that needs the
-     * transaction to be closed. Check if the url has an existing transaction and to further check
-     * if transaction is closed. If transaction is not found then return 404
+     * Pre handle method to validate the request before it reaches the controller by checking if
+     * transaction's status is closed.
      */
     @Override
     @SuppressWarnings("unchecked")
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
         Object handler) {
-        try {
-            Map<String, String> pathVariables = (Map) request
-                .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
-            String transactionId = pathVariables.get("transactionId");
-            ResponseEntity<Transaction> transaction = transactionManager
-                .getTransaction(transactionId, request.getHeader("X-Request-Id"));
-            request.setAttribute(AttributeName.TRANSACTION.getValue(), transaction.getBody());
-
-            return isTransactionClosed(transaction);
-
-        } catch (HttpClientErrorException httpClientErrorException) {
-            response.setStatus(httpClientErrorException.getStatusCode().value());
+        Transaction transaction = (Transaction) request
+            .getAttribute(AttributeName.TRANSACTION.getValue());
 
+        if (transaction == null) {
+            logRequestError(request);
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             return false;
         }
+        return TransactionStatus.CLOSED.getStatus().equals(transaction.getStatus());
     }
 
     /**
-     * Returns whether transaction is closed or not.
+     * Log error.
+     *
+     * @param request
      */
-    private boolean isTransactionClosed(ResponseEntity<Transaction> transaction) {
-        return (transaction.getBody() != null && transaction.getBody().getStatus()
-            .equals(TransactionStatus.CLOSED.getStatus()));
+    private void logRequestError(HttpServletRequest request) {
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("request_method", request.getMethod());
+        debugMap.put("message",
+            "ClosedTransactionInterceptor error: no transaction in request session");
```

Comment:
This message is specific to a certain case so shouldn't be in a generic method, given that this method is only used once and the calling method is now very short it should probably be done inline removing this method entirely

Reply:
I thought so. I'll change it